src/node_crypto.cc

```diff
@@ -4827,6 +4828,448 @@ void Scrypt(const FunctionCallbackInfo<Value>& args) {
 #endif  // OPENSSL_NO_SCRYPT
 
 
+class KeyPairGenerationConfig {
+ public:
+  virtual EVP_PKEY_CTX* Setup() = 0;
```

Comment:
Would it make sense to return a smart pointer? The methods currently all use `.release()`, and the only call site immediate turns it into a smart pointer again afaict?